charts/datadog/templates/agent-clusterchecks-cilium-network-policy.yaml

```diff
@@ -64,6 +67,18 @@ specs:
           - ports:
               - port: "5005"
                 protocol: TCP
+  - description: "Egress to kube-state-metrics"
+    endpointSelector:
+      matchLabels:
+        app: {{ template "datadog.fullname" . }}-clusterchecks
+    egress:
+      - toEndpoints:
+          - matchLabels:
+              name: kube-state-metrics
+        toPorts:
+          - ports:
+              - port: "8080"
+                protocol: TCP
```

Comment:
I would have the same remark as for the node agent: the cluster checks runners have [an “allow all egress traffic” rule here](https://github.com/DataDog/helm-charts/blob/7fb6a6ea7e079778d403ae284f3f1f45eaf3b1ac/charts/datadog/templates/agent-clusterchecks-cilium-network-policy.yaml#L67-L83).

We should rather fix this “allow all egress” rule rather than adding a `kube-state-metric` specific one.